Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/Allocator/Jobs/SendSearchRequestToDrivers/Handle/Internal/DriverPool.hs

```diff
@@ -206,9 +207,9 @@ prepareDriverPoolBatch cityServiceTiers merchant driverPoolCfg searchReq searchT
                   _ -> pure ([], [])
               logDebug $ "SpecialPickupZonePoolBatch goHomeDriversInQueue -" <> show goHomeDriversInQueue
               let goHomeDriversInQueueId = map (\d -> d.driverPoolResult.driverId) goHomeDriversInQueue
-              let normalDriversInQueue = filter (\d -> not (d.driverPoolResult.driverId `elem` goHomeDriversInQueueId)) driversInQueue
+              let normalDriversInQueue = filter (\d -> d.driverPoolResult.driverId `notElem` goHomeDriversInQueueId) driversInQueue
               let normalDriversInQueue' = bookAnyFilters transporterConfig normalDriversInQueue []
-              (_, normalDriversInQueueBatch) <- mkDriverPoolBatch merchantOpCityId_ (take driverPoolCfg.driverBatchSize normalDriversInQueue') intelligentPoolConfig transporterConfig batchSize False
+              (_, normalDriversInQueueBatch) <- mkDriverPoolBatch merchantOpCityId_ (take (driverPoolCfg.dynamicBatchSize V.! max (min ((length driverPoolCfg.dynamicBatchSize) - 1) (batchNum + 1)) 0) normalDriversInQueue') intelligentPoolConfig transporterConfig batchSize False
```

Comment:
What is this logic, can you make it as a common function?

Reply:
sure